web/packages/teleport/src/JoinTokens/JoinTokens.tsx

```diff
@@ -132,90 +142,130 @@ export const JoinTokens = () => {
+                  isSortable: false,
+                  render: renderRolesCell,
+                },
+                // expiryText is non render and used for searching
+                {
+                  key: 'expiryText',
+                  isNonRender: true,
+                },
+                // expiry is used for sorting, but we display the expiryText value
+                {
+                  key: 'expiry',
+                  headerText: 'Expires in',
+                  isSortable: true,
+                  render: ({ expiry, expiryText, isStatic, method }) => {
+                    const now = new Date();
+                    const isLongLived =
+                      isAfter(expiry, addHours(now, 24)) && method === 'token';
+                    return (
+                      <Cell>
+                        <Flex alignItems="center" gap={2}>
+                          <Text>{expiryText}</Text>
+                          {(isLongLived || isStatic) && (
+                            <HoverTooltip tipContent="Long-lived and static tokens are insecure and will be deprecated. Use short-lived tokens or alternative join methods (gcp, iam) for long-lived access.">
+                              <Warning size="small" color="error.main" />
+                            </HoverTooltip>
+                          )}
+                        </Flex>
+                      </Cell>
+                    );
+                  },
                 },
-              },
-              {
-                altKey: 'options-btn',
-                render: (token: JoinToken) => (
-                  <ActionCell
-                    token={token}
-                    onEdit={() => resources.edit(token.id)}
-                    onDelete={() => setTokenToDelete(token)}
-                  />
-                ),
-              },
-            ]}
-            emptyText="No active join tokens found"
-            pagination={{ pageSize: 30, pagerPosition: 'top' }}
-            customSearchMatchers={[searchMatcher]}
-            initialSort={{
-              key: 'expiry',
-              dir: 'ASC',
+                {
+                  altKey: 'options-btn',
+                  render: (token: JoinToken) => (
+                    <ActionCell
+                      token={token}
+                      onEdit={() => {
+                        // prefer editing in the standard form
+                        // if we support that join method
+                        if (
+                          token.method === 'iam' ||
+                          token.method === 'gcp' ||
+                          token.method === 'token'
+                        ) {
+                          setEditingToken(token);
+                          return;
+                        }
+                        // otherwise, edit in yaml editor
+                        resources.edit(token.id);
+                      }}
+                      onDelete={() => setTokenToDelete(token)}
```

Comment:
i think we are missing a mfa check with `delete` operation because i get a 403 forbidden with the error message (below).

also i think we are missing error handling? because the error was silent and I was not notified of an error (i just happened to have the networks tab opened and saw the error there). and despite error, the token still got deleted from the table making it seem like the delete succeeded, but on refresh the `deleted` token came back

<img width="543" alt="image" src="https://github.com/user-attachments/assets/93c53ae1-b48c-4a35-a73c-0c5e41dee0ed">

Reply:
Hmmm, it seems like we are missing delete error handling. I'll add that in, thank you. 

Delete might not be wrapped in the MFA stuff within our api code. I'll check